### PR TITLE
Pki - migrate to junit 5

### LINKED ISSUE #5560: pki - migrate tests from Junit4 to Junit5
Module pki
eg 
public abstract class BaseKeyStoreFileWrapperTest

- Add dependency for junit.jupiter (junit5)
- Migrate per https://blogs.oracle.com/javamagazine/post/migrating-from-junit-4-to-junit-5-important-differences-and-benefits
- Remove dependency on junit 4

Here's an example that migrates parameterized tests https://github.com/hyperledger/besu/pull/4975/files

### PR DESCRIPTION
**Pull request by macfarla**
fixes #5560

### CODE REVIEW

File: pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java [hunk 1]
```diff
@@ -158,7 +159,7 @@ public void getCertificateChain() throws Exception {
     when(keyStore.getCertificateChain(CERTIFICATE_ALIAS))
         .thenReturn(new Certificate[] {certificate});
 
-    assertThat(keyStoreWrapper.getCertificateChain(CERTIFICATE_ALIAS)).hasSize(1);
+    assertEquals(keyStoreWrapper.getCertificateChain(CERTIFICATE_ALIAS), 1);
```
Comment (jframe) [inline pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java:162]: This is comparing an array of Certificate with an int. The original assertion using hasSize is the correct thing to do

File: pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java [hunk 2]
```diff
@@ -171,7 +172,7 @@ public void getCertificateChainFallbackToTrustStore() throws Exception {
     when(trustStore.getCertificateChain(CERTIFICATE_ALIAS))
         .thenReturn(new Certificate[] {certificate});
 
-    assertThat(keyStoreWrapper.getCertificateChain(CERTIFICATE_ALIAS)).hasSize(1);
+    assertEquals(keyStoreWrapper.getCertificateChain(CERTIFICATE_ALIAS), 1);
```
Comment (jframe) [inline pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java:175]: This should be using hasSize or similar so we check the array size and not compare Certificate[] with an int

File: pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java [hunk 3]
```diff
@@ -188,10 +189,10 @@ public void loadKeyStoreFromFile() {
...
 
-    assertThat(loadedKeyStore.getPublicKey("validator")).isNotNull();
-    assertThat(loadedKeyStore.getPrivateKey("validator")).isNotNull();
-    assertThat(loadedKeyStore.getCertificate("validator")).isNotNull();
+    assertNotNull(loadedKeyStore.getPublicKey("validator"));
+    assertNotNull(loadedKeyStore.getPrivateKey("validator"));
+    assertNotNull(loadedKeyStore.getCertificate("validator"));
     // CA -> INTERCA -> PARTNERACA -> VALIDATOR
-    assertThat(loadedKeyStore.getCertificateChain("validator")).hasSize(4);
+    assertEquals(loadedKeyStore.getCertificateChain("validator"), 4);
```
Comment (jframe) [inline pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java:196]: This should be using hasSize or similar so we check the array size and not do a object comparison

File: pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java (hunk 1 above)
Comment (macfarla) [reply pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java:162]: fixed

File: pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java (hunk 2 above)
Comment (macfarla) [reply pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java:175]: fixed

File: pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java (hunk 3 above)
Comment (macfarla) [reply pki/src/test/java/org/hyperledger/besu/pki/keystore/SoftwareKeyStoreWrapperTest.java:196]: fixed

**Review by jframe — APPROVED** [review]
LGTM

**Pull request merged**